contrib/monitored_resource_util/src/main/java/io/opencensus/contrib/monitoredresource/util/MonitoredResourceUtils.java

```diff
@@ -50,5 +55,29 @@ public static MonitoredResource getDefaultResource() {
     return null;
   }
 
+  /**
+   * Returns a {@code Resource}. Detector sequentially runs resource detection from environment
+   * variables, GKE, GCP and AWS.
+   *
+   * @return a {@code Resource}.
+   * @since 0.18
+   */
+  @Nullable
+  public static Resource detectResource() {
+    List<Resource> resourceList = new ArrayList<Resource>();
+    resourceList.add(Resource.createFromEnvironmentVariables());
+
+    if (System.getenv("KUBERNETES_SERVICE_HOST") != null) {
+      resourceList.add(GcpGkeContainerMonitoredResource.createResource());
+    }
+    if (GcpMetadataConfig.getInstanceId() != null) {
```

Comment:
This should be an `else if`. GKE containers also has instance id.

Reply:
```mergeResources``` API merges all resources results and in case a type of label key is already set, the first set value takes precedence.

Reply:
Let me know if you still want me to make that change.

Reply:
I am confused about the order of the resources in the list. Also why do we merge all of them?

Reply:
> I am confused about the order of the resources in the list.

Population from environment variables MUST be the first applied detection process unless the user explicitly overwrites this behavior. @fabxc Could you please confirm order of the other resources in the list.

> why do we merge all of them?

This is mentioned in the Specs : https://github.com/census-instrumentation/opencensus-specs/blob/master/resource/Resource.md#merging

@fabxc Could you please add more, if I missed anything.

Reply:
For this one I would still suggest to pick up either GKE or GCE, since one instance cannot be both a GCE instance and a GKE instance (correct me if I'm wrong).

Reply:
Done